src/main/typescript/rug/operations/Decorators.ts

```diff
@@ -27,137 +30,190 @@ function get_metadata(obj: any, key: string) {
-  "command-handler",
-  "handle must be a function with first parameter = HandlerContext");
-const ResponseHandler = ruglike(
-  "handle",
-  "response-handler",
-  "handle must be a function with first parameter = Response<T>");
+function declareRug(obj: any, kind: RugKind, description: string, name?: string) {
+  if (name === undefined) {
+    set_metadata(obj, "__description", description);
+  } else {
+    set_metadata(obj, "__description", description);
+    set_metadata(obj, "__name", name);
+  }
+  set_metadata(obj, "__kind", kind);
+}
 
-const EventHandler = (
+export function EventHandler(
   nameOrDescription: string,
   descriptionOrExpression: string | PathExpression<any, any>,
-  expression?: PathExpression<any, any> | string) => {
-  // tslint:disable-next-line:ban-types
-  return (ctr: Function) => {
-    if (typeof ctr.prototype.handle !== "function") {
-      throw new Error("handle must be a function with first parameter = Match<R,N>");
-    }
+  expression?: PathExpression<any, any> | string) {
+  return (obj: object) => {
     if (expression !== undefined) {
-      set_metadata(ctr.prototype, "__name", nameOrDescription);
-      set_metadata(ctr.prototype, "__description", descriptionOrExpression);
-      set_metadata(ctr.prototype, "__kind", "event-handler");
-      if (typeof expression === "string") {
-        set_metadata(ctr.prototype, "__expression", expression);
-      } else {
-        set_metadata(ctr.prototype, "__expression", expression.expression);
-      }
-    } else {
-      set_metadata(ctr.prototype, "__description", nameOrDescription);
-      set_metadata(ctr.prototype, "__kind", "event-handler");
       if (typeof descriptionOrExpression === "string") {
-        set_metadata(ctr.prototype, "__expression", descriptionOrExpression);
+        declareEventHandler(obj, descriptionOrExpression, expression, nameOrDescription);
       } else {
-        set_metadata(ctr.prototype, "__expression", descriptionOrExpression.expression);
+        throw Error("When three parameters are passed, the second must be a description");
       }
+    } else {
+      declareEventHandler(obj, nameOrDescription, descriptionOrExpression);
     }
   };
-};
+}
+
+export function declareEventHandler(
+  obj: any, description: string, expression: string | PathExpression<any, any>, name?: string) {
+  declareRug(obj, "event-handler", description, name);
+  if (typeof expression === "string") {
+    set_metadata(obj, "__expression", expression);
+  } else {
+    set_metadata(obj, "__expression", expression.expression);
+  }
+  return obj;
+}
 
 /**
  * Decorator for tags. Sets tags on the class
  */
 
-function Tags(...tags: string[]) {
+export function Tags(...tags: string[]) {
   return (target: any) => {
-    set_metadata(target.prototype, "__tags", tags);
+    declareTags(target, tags);
   };
 }
 
-function Intent(...intent: string[]) {
+export function declareTags(target: any, tags: string[]) {
+  set_metadata(target, "__tags", tags);
```

Comment:
Should we append to the existing tags metadata to allow it be invoked multiple times?